### PR TITLE
Logging unattached items

### PR DESCRIPTION
**Pull request by fredex42**
logging items that are not found in asset management to dynamodb

### CODE REVIEW

**Comment by tomrf1** [comment]
I still think actors add unnecessary complexity to this project but the code looks fine :)
👍

File: src/main/scala/actors/UnattachedAtomActor.scala [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+          case Some(response)=>response match {
+            case Left(error)=>
+              logger.error(s"Unable to record missing item: $error")
+              if(attempt>=maxRetries){
+                logger.error(s"Not able to record after $maxRetries attempts, giving up.")
+                sender ! ErrorSend(s"Not able to record after $maxRetries attempts, giving up.")
+              } else {
+                //retry after a short delay
+                Thread.sleep(2000)
+                self ! MasterNotFound(atomId, createdChange, updatedChange, attempt + 1)
```
Comment (davidfurey) [inline src/main/scala/actors/UnattachedAtomActor.scala]: If this retry logic is executed I think the origSender will not be notified

File: src/main/scala/actors/UnattachedAtomActor.scala [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        case Success(maybeResult)=>maybeResult match {
+          case None=> //we expect no data back from an insert
+            logger.info("Saved unattached atom to table")
+            origSender ! SuccessfulSend
+          case Some(response)=>response match {
+            case Left(error)=>
+              logger.error(s"Unable to record missing item: $error")
+              if(attempt>=maxRetries){
+                logger.error(s"Not able to record after $maxRetries attempts, giving up.")
+                sender ! ErrorSend(s"Not able to record after $maxRetries attempts, giving up.")
```
Comment (davidfurey) [inline src/main/scala/actors/UnattachedAtomActor.scala]: Should this be `origSender`?

**Comment by fredex42** [comment]
@tomrf1 yeah if I get time i will re-examine the necessity for actors, but need to get it working better first :)

File: src/main/scala/actors/UnattachedAtomActor.scala (hunk 2 above)
Comment (fredex42) [reply src/main/scala/actors/UnattachedAtomActor.scala]: you're absolutely right, thanks

File: src/main/scala/actors/UnattachedAtomActor.scala (hunk 1 above)
Comment (fredex42) [reply src/main/scala/actors/UnattachedAtomActor.scala]: hmmm, that's right. I don't think that's a big problem as origSender is only really used for testing, to ensure that it processes the messages correctly, but it's not very neat I suppose

Comment (fredex42) [reply src/main/scala/actors/UnattachedAtomActor.scala]: fixed it in next commit

**Pull request merged**